tor/core/mentions.py

```diff
@@ -35,11 +38,20 @@ def process_mention(mention, r, tor, redis_server, config):
         parent.title = 'Unknown Content'
     else:
         # this is a post.
-        parent = r.submission(id=clean_id(mention.parent_id))
+        parent = submission
         parent_permalink = parent.permalink
         # format that sucker so it looks right in the template.
         parent.title = '"' + parent.title + '"'
 
+    # Ignore requests made by the OP of content or the OP of the submission
+    if mention.author.id == parent.author.id or \
+            mention.author.id == submission.author.id:
```

Comment:
I found a property `link_id` in `Comment` on [here](https://github.com/reddit/reddit/wiki/JSON) which seems to do the exact same thing (well, I obviously can't test the bot but from playing with PRAW in a shell it seems to work, and [this](https://www.reddit.com/r/redditdev/comments/29ewkl/praw_get_submissionid_of_a_comment/cikcydx/) says the same). I'll use that for now, seems a bit strange to me that nested comments wouldn't have a direct reference to the submission they're in.

Comparing author properties seems to work as well yes, they are different instances but seem to overload the equality operator.